### ISSUE TITLE
Possibly a false positive for UP007

### ISSUE THREAD

**Issue by rassie** [charliermarsh/ruff#1560]
Here is a bit of code I adapted [from StackOverflow](https://stackoverflow.com/questions/61753056/partial-update-in-fastapi/71135795#71135795) to make Pydantic model's fields optional:

```python
def convert_to_optional(schema: type[Base]) -> dict:
    return {k: Optional[v] for k, v in schema.__annotations__.items()} 
```

This triggers UP007, proposing to replace `Optional[v]` with `v | None`, which the compiler denies (`TypeError: unsupported operand type(s) for |: 'str' and 'NoneType'`). Since `v` is a value and not a type, I suppose this is a false positive.

That being said: I consider this issue edge-casey and thus simply informative, both for developers and people searching the web for the problem. I'm fully  content with just adding `# noqa: UP007` to that line and be done with it.

**Comment by charliermarsh** [comment]
Let me see if pyupgrade handles this case correctly.

**Comment by charliermarsh** [comment]
I think this is actually just a bug.

**Comment by charliermarsh** [comment]
Because `|` is never valid in a runtime context like that.

**Comment by charliermarsh** [comment]
It looks like pyupgrade avoids these, presumedly for this reason.
